src/components/ha-selector/ha-selector-datetime.ts

```diff
@@ -39,13 +39,13 @@ export class HaDateTimeSelector extends LitElement {
           .locale=${this.hass.locale}
           .disabled=${this.disabled}
           .required=${this.required}
-          .value=${values?.[0]}
+          .value=${values?.[0] === "undefined" ? undefined : values?.[0]}
```

Comment:
The default value for date picker is undefined. If toggle to yaml mode in this case, we get a value of `undefined 00:00:00`. 

If immediately toggle back to UI mode, this throws an exception and crash the date picker, because `"undefined"` is not parseable as a date. This handles that case better.

Reply:
would it be the string undefined? Then we should probably check/fix where it is set to that.

Reply:
Yes it is string undefined. 

When we run this:

```
    fireEvent(this, "value-changed", {
      value: `${this._dateInput.value} ${this._timeInput.value}`,
    });
```

If this._dateInput.value is undefined, this sets value to "undefined 00:00:00"

Then in render, we split value:
```
const values = this.value?.split(" ");
```

And so we set date value to "undefined", and time value to "00:00:00". And date picker throws an error because "undefined" is not a date (it seems happy with value=undefined, in which case it just leaves the field blank).

Reply:
we should catch that when we set the value

Reply:
Catch it, and do what with it exactly?

Instead of `undefined 00:00:00`, should the picker value just be `00:00:00` by default? If that's the case I will still need extra handling before `split` in render to recognize that the date is missing. But I can do this if you prefer to not see literal "undefined" in the yaml. 

Other idea is to just make the date always valid by pre-populating it in the same way that time is pre-populated with 12am. It could just always start at todays date (or 1/1/1970?) and then it would never be undefined. Let me know if you would prefer that. But that doesn't seem to match what is done elsewhere.

Reply:
I updated to use a different approach instead of handling "undefined" as a string. Let me know if you like that better.